### PR TITLE
safely skip getting roles for sandboxes that are broken

### PR DESCRIPTION
**Pull request by malaffoon**

### CODE REVIEW

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/sandbox/DefaultSandboxService.java [hunk 1]
```diff
@@ -39,8 +44,8 @@ public DefaultSandboxService(final SandboxRepository sandboxRepository,
                 .comparing(sandbox -> Optional.ofNullable(sandbox.getLabel()).orElse(""));
 
         return sandboxes.stream()
-                .map(tenant ->
-                        Sandbox.builder()
+                .filter(tenant -> rolesBySandboxKey.containsKey(tenant.getKey()))
```
Comment (malaffoon) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/sandbox/DefaultSandboxService.java]: If there are no roles for a sandbox it seems pointless to return it; however, i defer to @ftdholbrook or @esotrope if they feel we can/should return the sandbox with no roles.

Comment (ftdholbrook) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/sandbox/DefaultSandboxService.java]: I think that is correct.   Really we should get the sandbox list from the admin service, not the way we are doing it here.  we could decorate it here with the roles.

Comment (ftdholbrook) [reply webapp/src/main/java/org/opentestsystem/rdw/reporting/sandbox/DefaultSandboxService.java]: ... I guess the only down side is when you are waiting on a migration, that's the only sign we have right now that is what is going on.

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/sandbox/DefaultSandboxService.java [hunk 2]
```diff
@@ -58,12 +63,16 @@ public DefaultSandboxService(final SandboxRepository sandboxRepository,
...
-            final List<SandboxRole> roles = sandboxRoleRepository.findAll();
-            TenantContextHolder.clear();
-            rolesBySandboxKey.put(tenant.getKey(), roles);
-        });
+        for (final Tenant sandbox : sandboxes) {
+            TenantContextHolder.setTenantId(sandbox.getId());
+            try {
+                rolesBySandboxKey.put(sandbox.getKey(), sandboxRoleRepository.findAll());
+            } catch (final Exception e) {
+                logger.warn("Failed to retrieve sandbox roles for {}: {}", sandbox.toString(), e.getMessage());
```
Comment (esotrope) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/sandbox/DefaultSandboxService.java]: does this happen after filtering?

**Pull request merged**